content/tutorials/flutter-cryptocurrency-app.textile

```diff
@@ -0,0 +1,720 @@
+
+Building the chat room is fairly simple, and with Ably realtime capability, it works like a charm with little code.
+
+Back to @AblySevice@, we will add two methods, one to listen to the latest messages as long as a user is inside the chat room, and the other is to send new messages.
+
+```[flutter]
+ChatUpdates getChatUpdates() {
+  ChatUpdates _chatUpdates = ChatUpdates();
+
+  _chatChannel = _realtime.channels.get('public-chat');
+
+  var messageStream = _chatChannel.subscribe();
+
+  messageStream.listen((message) {
+    _chatUpdates.updateChat(
+      ChatMessage(
+        content: message.data,
+        dateTime: message.timestamp,
+        isWriter: message.name == "${_realtime.clientId}",
+      ),
+    );
+  });
+
+  return _chatUpdates;
+}
+```
+
+Channel names can be chosen almost freely. All names are always only valid for one specific app, so channels of different apps can have the same name (if you want to send messages from one app to the other you can do this by using an API Streamer). For the chat we will use @public-chat@. The name of the channel has to be the same when sending and receiving messages. If that’s not the case this would mean we are listening to a different channel than the one we publish to, so no messages will be received.
+
+If the channel doesn’t exist at the time this function is called, it will be created as a new one automatically.
+
+Just like we did previously with prices, we will create @ChatUpdates@ as a @ChangeNotifier@ holding the most recently published message and push it to a queue in the UI. By calling @getChatUpdates()@ on the same instance of @AblyService@ registered previously with get_it, listening on the chat channel will be started and each new message arrives will be delivered to the listeners.
+
+For sending messages we will add @sendMessage()@ to the service.
+
+```[flutter]
+Future sendMessage(String content) async {
+  _realtime.channels.get('public-chat');
+
+  await _chatChannel.publish(data: content, name: "${_realtime.clientId}");
+}
+```
+
+To publish messages, we will call @publish@ with the name of the channel, and it will send the messages to it. The @name@ parameter is optional, it can be used to differentiate different types of messages that are sent over the same channel. We defined it to be the current client ID of the connected device, so we can differentiate the messages sent by the current user from other messages of other users on the chat channel.
+
+To not add unnecessary code for this demo app we don’t store the client ID which means it will change every time you start the app.
+
+With this our chat infrastructure in the service is done. So lets move to the chat view. Once we open it, we need it to initialize a listener, just like we did on the main page @DashboardView@.
+
+In @chat.dart@ we will initialize the channel and set up our listener:
```

Comment:
They have to create this file first. I know its a bit annoying but the should be lead by the hand